File: app/src/main/java/br/com/jonathanzanella/myexpenses/receipt/ReceiptContract.java
package br.com.jonathanzanella.myexpenses.receipt;

import android.content.Context;
import android.support.annotation.StringRes;

import org.joda.time.DateTime;

import br.com.jonathanzanella.myexpenses.account.Account;
import br.com.jonathanzanella.myexpenses.source.Source;
import br.com.jonathanzanella.myexpenses.validations.ValidationError;

interface ReceiptContract {
	interface View {
		Context getContext();
		void setTitle(@StringRes int string);
		void setTitle(String string);
		void showReceipt(Receipt receipt);
	}

	interface EditView extends View {
		Receipt fillReceipt(Receipt receipt);
		void finishView();
		void showError(ValidationError error);
		void onSourceSelected(Source source);
		void onAccountSelected(Account account);
		int getInstallment();
		int getRepetition();
		void onDateChanged(DateTime balanceDate);
	}
}

File: app/src/main/kotlin/br/com/jonathanzanella/myexpenses/receipt/ReceiptContract.kt
package br.com.jonathanzanella.myexpenses.receipt

import android.content.Context
import android.support.annotation.StringRes
import br.com.jonathanzanella.myexpenses.account.Account
import br.com.jonathanzanella.myexpenses.source.Source
import br.com.jonathanzanella.myexpenses.validations.ValidationError
import org.joda.time.DateTime

interface ReceiptContract {
    interface View {
        val context: Context
        fun setTitle(@StringRes string: Int)
        fun setTitle(string: String)
        fun showReceipt(receipt: Receipt)
    }

    interface EditView : View {
        fun fillReceipt(receipt: Receipt): Receipt
        fun finishView()
        fun showError(error: ValidationError)
        fun onSourceSelected(source: Source)
        fun onAccountSelected(account: Account)
        val installment: Int
        val repetition: Int
        fun onDateChanged(balanceDate: DateTime)
        fun showConfirmDialog(receipt: Receipt)
    }
}
